gst/parser/gstttmlparse.c

```diff
@@ -1653,15 +1656,17 @@ handle_buffer (GstTtmlParse * self, GstBuffer * buf)
           gst_pad_push_event (self->srcpad, event);
         }
 
+        free_subtitles (subtitle_list);
+        destroy_subs_parser (subs_parser);
         return GST_FLOW_OK;
       }
 
       if (ret != GST_FLOW_OK)
         GST_DEBUG_OBJECT (self, "flow: %s", gst_flow_get_name (ret));
     }
 
-    g_list_free (subtitle_list);
-    destroy_subs_parser(subs_parser);
+    free_subtitles (subtitle_list);
```

Comment:
This seems to be the culprit, If I commit it out I get subtitles again.

Reply:
Actually I made a mistake when I commented, the `free_subtitles` I meant is at line 1659.

Reply:
hm, it's weird because I definitely see subtitles
![image](https://user-images.githubusercontent.com/22928774/200854917-c3bbaa7f-4add-4ea2-affe-aeed8f869ceb.png)
But it's helloworld runned under Win. Will try it under Linux tomorrow...

Reply:
Works good too
![image](https://user-images.githubusercontent.com/22928774/200856298-af864205-bbef-4980-b651-64e843f97ba8.png)

Reply:
Strange, I don't get a single cue unless I comment the line I mentioned out.

Reply:
Yeah, no subtitles and crash after few seconds in tplay. Will see on it tomorrow.

Reply:
Yes, you right. As I understand after `gst_pad_push` call (L1649) we loosing control on buffer and gstreamer should free it by itself (and `GstMetaInfo` as well) but it just doesn't happens on some reason.

Reply:
I mean, the SubtitleParser does keep the buffers as well, so if you did it before you started calling `destroy_subs_parser()`, maybe that is it?

Reply:
I created an experimental draft PR with some things I found. It makes some fixes to the parser loop and makes sure we copy certain length expression instances to avoid double-freeing them when we release the buffers.

https://github.com/castlabs/gst-ttml-subtitles/pull/16